Project/DAQ_System/Program/Src/Sensor/GyroScope/l3gd20h.cpp

```diff
@@ -0,0 +1,150 @@
+
+#include "l3gd20h.hpp"
+#include <stdint.h>
+
+#include "i2c.h"
+extern I2C_HandleTypeDef hi2c1;
+
+
+
+Gyroscope_L3GD20H::Gyroscope_L3GD20H(){
```

Comment:
Consider requiring the `hi2c1` object to be a parameter of the class constructor. This is called "dependency injection".

You can look at the linpots for a quick example:
* [Included the auto-generated](https://github.com/DallasFormulaRacing/DataAcquisition2.0/blob/develop/Project/DAQ_System/Program/Src/Sensor/LinearPotentiometer/sls1322.hpp#L17) ADC header file
* [Declared an ADC object](https://github.com/DallasFormulaRacing/DataAcquisition2.0/blob/develop/Project/DAQ_System/Program/Src/Sensor/LinearPotentiometer/sls1322.hpp#L58) in the header file.
* [Required the object to be passed through the constructor](https://github.com/DallasFormulaRacing/DataAcquisition2.0/blob/develop/Project/DAQ_System/Program/Src/Sensor/LinearPotentiometer/sls1322.cpp#L18) and initialized it to the object we declared. The part after the `:` is the "initialization list".
* Then, in practice, the linpot is [initialized like this](https://github.com/DallasFormulaRacing/DataAcquisition2.0/blob/develop/Project/DAQ_System/Program/Src/app.cpp#L94).